### PR TITLE
Point Texture to the latest master branch fixes + LessonController improvements

### PR DESCRIPTION
**Pull request by ffizzik**
We had a random crash on LessonController, so I made a few small improvements and set the Texture SDK to use the master branch. This is because after their last release version 3.1.0 in September 2021, they accepted some changes, including one for Xcode 15, but didn't release a new version. 

I did some testing on my end and seems like it's not crashing anymore. Would be super grateful if someone could test it as well

### CODE REVIEW

File: Sabbath School/Lesson/Controller/LessonController.swift [hunk 1]
```diff
@@ -36,9 +36,10 @@ enum LessonControllerSections: Int {
 }
 
 final class LessonController: ASDKViewController<ASDisplayNode> {
-    var tableNode: ASTableNode? { return node as? ASTableNode }
-    
-    var delegate: LessonControllerDelegate?
+    var tableNode: ASTableNode {
+           return node as! ASTableNode
+    }
+    weak var delegate: LessonControllerDelegate?
```
Comment (EmersonCarpes) [inline Sabbath School/Lesson/Controller/LessonController.swift:41]: I don't think force unwrap is a good idea

**Review by VitalikL — APPROVED** [review]
LGTM!

**Review by EmersonCarpes — APPROVED** [review]

**Comment by EmersonCarpes** [comment]
@ffizzik could you update version to to 3.5.1 Build 1 please?

**Comment by ffizzik** [comment]
> @ffizzik could you update version to to 3.5.1 Build 1 please?

sure

**Pull request merged**